h2/src/main/org/h2/mvstore/Cursor.java

```diff
@@ -13,42 +13,76 @@
  * @param <K> the key type
  * @param <V> the value type
  */
-public class Cursor<K, V> implements Iterator<K> {
-
-    private final MVMap<K, ?> map;
-    private final K from;
-    private CursorPos pos;
-    private K current, last;
-    private V currentValue, lastValue;
+public final class Cursor<K, V> implements Iterator<K> {
+    private final K to;
+    private CursorPos cursorPos;
+    private CursorPos keeper;
+    private K current;
+    private K last;
+    private V lastValue;
     private Page lastPage;
-    private final Page root;
-    private boolean initialized;
 
-    Cursor(MVMap<K, ?> map, Page root, K from) {
-        this.map = map;
-        this.root = root;
-        this.from = from;
+    public Cursor(Page root, K from) {
+        this(root, from, null);
+    }
+
+    public Cursor(Page root, K from, K to) {
+        this.cursorPos = traverseDown(root, from);
+        this.to = to;
     }
 
     @Override
+    @SuppressWarnings("unchecked")
     public boolean hasNext() {
```

Comment:
Could you tell please, what's the idea behind the code in that method? I'm trying to get to know how it's done but apparently I can't do it without a clue